Sources/StreamChat/Repositories/SyncRepository.swift

```diff
@@ -108,10 +137,76 @@ class SyncRepository {
                 }
                 return
             }
-
-            self?.syncLocalState(lastSyncAt: lastSyncAt, completion: completion)
+            if self?.usesV2Sync == true {
+                self?.syncLocalStateV2(lastSyncAt: lastSyncAt, completion: completion)
+            } else {
+                self?.syncLocalState(lastSyncAt: lastSyncAt, completion: completion)
+            }
         }
     }
+    
+    // MARK: - V2
+    
+    /// Runs offline tasks and updates the local state for channels
+    ///
+    /// Recovery mode (pauses regular API requests while it is running)
+    /// 1. Enter recovery
+    /// 2. Runs offline API requests
+    /// 3. Exit recovery
+    ///
+    /// Background mode (other regular API requests are allowed to run at the same time)
+    /// 1. Collect all the **active** channel ids (from instances of `Chat`, `ChannelList`, `ChatChannelController`, `ChatChannelListController`)
+    /// 2. Apply updates from the /sync endpoint for these channels
+    /// 3. Refresh channel lists (channels for current pages in `ChannelList`, `ChatChannelListController`)
+    private func syncLocalStateV2(lastSyncAt: Date, completion: @escaping () -> Void) {
```

Comment:
We also have some internal docs about the sync mechanism, that probably require revisiting.

Reply:
It is still there, that logic is in `OfflineRequestsRepository` what did not change.